### DISCUSSION TITLE
[Feature Request]:  "No Channel" and "Unknown Public Key" error by direct message

### DISCUSSION THREAD

**Discussion by jirogit** [meshtastic/firmware#5122] [Ideas]
### Platform

ESP32

### Description

I did "Full Erase and Install" firmware ver 2.5.7 to Lilygo T-LoRa T3-S3. 

I sent direct messages from Teltec V3 and RAK 4631.

I got "no channel" error.

I sent direct message from Lilygo T-LoRa T3-S3 to other nodes, I got "unknown encryption" error.

I needed to delete Lilygo T-LoRa T3-S3 node information from other nodes to resolve this issue.

I used iOS iPhone app ver 2.5.9.

My request is some other way to resolve this issue without deleting node information from other nodes manually.

**Comment by jirogit** [comment]
Teltec V3 ---> Heltec V3

"unknown encryption" error ---> "Unknown Public Key" error.
"no channel" error ---> "No Channel" error.

I also noticed "Encrypted Send Failed" When I updated firmware 2.5.7 ---> 2.5.8 from updated node sending time.

**Reply by lagunacomputer** [reply]
delete the nodes and wait

**Comment by lagunacomputer** [comment]
yeah we need a button that does "Manually Send/Exchange/Refresh Keys" either to a targeted node, or the whole mesh (maybe a within x hop limit?).   its causing frustrations having to manually delete nodes/nodedb and wait everytime we update fw on each node. At the least, each test send is causing lots of dead end TX's traffic on the band until the nodes sort it out

**Reply by garthvh** [reply]
You caused this with the full erase, going forward if you don't want to delete nodes then don't do a full erase when flashing. The warning and node deletion is a necessary part for this to be secure.

**Reply by lagunacomputer** [reply]
I build and flash from github code because of diy hardware.  Not sure how to choose that option from device-install.bat

**Comment by jirogit** [comment]
I went to Meshtastic phone app "setting" tab --- Security and copied the Public Key and Private Key.

I updated the firmware with full erase.
I sent a direct message. It gave me  "Encrypted Send Failed" error.

I copied back both the Private Key and the Public Key.

It worked. I was able to send the direct message successfully.

I felt this "key saving & restore" action  was easier than "deleting from other nodes list" action.

Maybe copying the Public Key and Private Key and putting these back to the same device can be done by the new phone app function?

**Reply by garthvh** [reply]
> I went to Meshtastic phone app "setting" tab --- Security and copied the Public Key and Private Key.
> 
> I updated the firmware with full erase. I sent a direct message. It gave me "Encrypted Send Failed" error.
> 
> I copied back both the Private Key and the Public Key.
> 
> It worked. I was able to send the direct message successfully.
> 
> I felt this "key saving & restore" action was easier than "deleting from other nodes list" action.
> 
> Maybe copying the Public Key and Private Key and putting these back to the same device can be done by the new phone app function?

If you don't want to wipe the keys don't do a full erase.

**Comment by LeeMaisel** [comment]
Is there a fix to this?  I have several devices, repeaters on remote hills that are inaccessible for the most part which makes it very hard to reset them.

**Reply by mrbreitkreutz** [reply]
Have the same issue and I haven't found a way to fix this except get every device to individually be reset to remove the stored keys
